### PR TITLE
add syntax and lint checks

### PR DESCRIPTION
**Pull request by dcrissman**
these are fairly standard puppet checks we should be running.

### CODE REVIEW

File: manifests/client/configure.pp [hunk 1]
```diff
@@ -35,8 +34,6 @@
     $lbclient_cert_file_path = undef,
     $lbclient_cert_password = undef,
     $lbclient_cert_alias = undef,
-    $before = [],
```
Comment (jewzaam) [inline manifests/client/configure.pp]: This seems out of place for this PR...

Comment (dcrissman) [inline manifests/client/configure.pp]: That line causes the syntax check to fail. Apparently $before in that context is implicit and so declaring it like this is unnecessary.

**Comment by alechenninger** [comment]
Will the linter error at anything or does it only give warnings?

**Comment by dcrissman** [comment]
@alechenninger The syntax check failed the build, so I assume a Lint failure would do the same. I did not see Lint warnings printed.

Take a look at my earlier noop PR, in the second commit you can see what the syntax error looked like. The third commit included Lint, but with very little output.
https://github.com/lightblue-platform/lightblue-puppet/pull/85

**Comment by alechenninger** [comment]
@dcrissman Ah, I see. Yeah, I was running against my PR's to see if there were any problems. Saw a ton of lint warnings but no errors, so was curious what the error rules were if you knew.

**Comment by dcrissman** [comment]
Failure should be fixed once #89 is merged.

**Comment by dcrissman** [comment]
Ready for review

File: spec/classes/lightblue/application/migrator_spec.rb [hunk 2]
```diff
@@ -40,9 +40,9 @@
         }
       )
       
-      should contain_file(client_config) \
-        .that_comes_before("Service[#{service_name}]") \
-        .that_notifies("Service[#{service_name}]")
+      should contain_lightblue__client__configure(client_config) \
```
Comment (jewzaam) [inline spec/classes/lightblue/application/migrator_spec.rb]: There is a few thinks that I don't understand here:
* what is `contain_lightblue__client__configure` and where is it defined?
* if it should notify a service, what is the "comes before" bit afterwards?

Comment (dcrissman) [inline spec/classes/lightblue/application/migrator_spec.rb]: >* what is contain_lightblue__client__configure and where is it defined?

This is when you want to test that a class (or define as in this case) has been included by the class being tested. When '::' are used, as in lightblue::client::configure, then for testing purposes you need to replace the colons with underscores.
see: http://rspec-puppet.com/matchers/

>*if it should notify a service, what is the "comes before" bit afterwards?
You make an excellent point. A new commit to come.

**Pull request merged**